xunit.runner.visualstudio.desktop/VsTestRunner.cs

```diff
@@ -445,12 +459,83 @@ void RunTests(IRunContext runContext, IFrameworkHandle frameworkHandle, LoggerHe
 #endif
 
                 var diagnosticSink = new DiagnosticMessageSink(logger, assemblyDisplayName, runInfo.Configuration.DiagnosticMessagesOrDefault);
-                using (var controller = new XunitFrontController(appDomain, assemblyFileName: assemblyFileName, configFileName: null, shadowCopy: shadowCopy, diagnosticMessageSink: MessageSinkAdapter.Wrap(diagnosticSink)))
+                var diagnosticMessageSink = MessageSinkAdapter.Wrap(diagnosticSink);
+                using (var controller = new XunitFrontController(appDomain, assemblyFileName: assemblyFileName, configFileName: null, shadowCopy: shadowCopy, diagnosticMessageSink: diagnosticMessageSink))
                 {
-                    var xunitTestCases = runInfo.TestCases.Select(tc => new { vs = tc, xunit = Deserialize(logger, controller, tc) })
```

Comment:
Issue: in case of Run All; there was an extra cost of `Serialize` after discovery and `Deserialize` before execution.

Solution: remove deserialize and make serialize conditional based on `RequireXunitTestProperty`.